### ISSUE TITLE
Compound Assignment not Handled Correctly

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#128]
```
The compound assignment (e.g. "a += b") does not appear to be treated 
correctly.  In particular, the checker does not issue a warning if a 
"non-regex" string is concatenated with a @Regex string, as in the following 
example:

  @Regex String a = "a";
  a += "(";

A full example is attached, and the checker is expected to issue a warning (but 
doesn't).
```

Original issue reported on code.google.com by `stefanheule@gmail.com` on 16 Feb 2012 at 3:15

Attachments:
- [TestRegex.java](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-128/comment-0/TestRegex.java)
